dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -967,11 +966,29 @@ public List<BlockWorkerInfo> getAllWorkers() throws IOException {
       List<BlockWorkerInfo> lostWorkers = masterClientResource.get().getLostWorkerInfoList()
           .stream().map(w -> new BlockWorkerInfo(w.getAddress(), w.getCapacityBytes(),
               w.getUsedBytes(), false)).collect(toList());
-      List<BlockWorkerInfo> allWorkers = new ArrayList<>();
-      allWorkers.addAll(liveWorkers);
-      allWorkers.addAll(lostWorkers);
-      return allWorkers;
+      // avoid duplicate elements in list
+      return combineAllWorkers(liveWorkers, lostWorkers);
+    }
+  }
+
+  private List<BlockWorkerInfo> combineAllWorkers(List<BlockWorkerInfo> liveWorkers,
+                                                  List<BlockWorkerInfo> lostWorkers) {
+    List<BlockWorkerInfo> allWorkers = new ArrayList<>();
+    for (BlockWorkerInfo liveWorker : liveWorkers) {
+      boolean isActive = true;
+      for (BlockWorkerInfo lostWorker : lostWorkers) {
+        if (lostWorker.getNetAddress().dumpMainInfo()
```

Comment:
Just realize this is comparing with NetAddress, is the worker id ready for use?
Eventually we want the worker id compared.

Reply:
Worker id is in @dbw9580's PR, but now we don't have worker id information to do so. I think @dbw9580 needs to change this code when he checks in his PR.

Reply:
Then let bowen change this part once his change is in.

Reply:
Noting the fact that it is reasonable to assume that there are only a few elements in the lostWorkerList.